packages/sdk/snet/sdk/mpe_contract.py

```diff
@@ -0,0 +1,74 @@
+import base64
+
+import web3
+from snet.sdk.payment_channel import PaymentChannel
+
+from snet.snet_cli.utils import get_contract_object, get_contract_deployment_block
+
+
+BLOCKS_PER_BATCH = 20000
+
+
+class MPEContract:
+    def __init__(self, w3):
+        self.web3 = w3
+        self.contract = get_contract_object(self.web3, "MultiPartyEscrow.json")
+        self.event_topics = [self.web3.sha3(text="ChannelOpen(uint256,uint256,address,address,address,bytes32,uint256,uint256)").hex()]
+        self.deployment_block = get_contract_deployment_block(self.web3, "MultiPartyEscrow.json")
+
+
+    def get_past_open_channels(self, account, service, starting_block_number=0, to_block_number=None):
```

Comment:
Why this function is so complicated? Why do you read logs in batches (while loop)??? Please take a look at ```MPEChannelCommand._get_all_filtered_channels``` in ```mpe_channel_command.py```. It works perfectly without while loop. You need to combine ```_get_all_channels_filter_sender_recipient_group``` and ```_get_all_filtered_channels```